src/PlcServer.cs

```diff
@@ -43,7 +45,209 @@ public PlcServer(OpcPlcConfiguration config, PlcSimulation plcSimulation, TimeSe
+            throw;
+        }
+    }
+
+    public override ResponseHeader CreateMonitoredItems(
+        RequestHeader requestHeader,
+        uint subscriptionId,
+        TimestampsToReturn timestampsToReturn,
+        MonitoredItemCreateRequestCollection itemsToCreate,
+        out MonitoredItemCreateResultCollection results,
+        out DiagnosticInfoCollection diagnosticInfos)
+    {
+        try
+        {
+            OperationContext context = ValidateRequest(requestHeader, RequestType.CreateSubscription);
+
+            var responseHeader = base.CreateMonitoredItems(requestHeader, subscriptionId, timestampsToReturn, itemsToCreate, out results, out diagnosticInfos);
+
+            Meters.AddMonitoredItemCount(context.SessionId.ToString(), subscriptionId.ToString(), itemsToCreate.Count);
+
+            _logger.LogDebug("{function} completed successfully with sessionId: {sessionId}, subscriptionId: {subscriptionId} and count: {count}.",
+                nameof(CreateMonitoredItems),
+                context.SessionId,
+                subscriptionId,
+                itemsToCreate.Count);
+
+            return responseHeader;
+        }
+        catch (Exception ex)
+        {
+            Meters.RecordTotalErrors(nameof(CreateMonitoredItems), ex.GetType().ToString());
+            _logger.LogError(ex, "Error creating monitored items");
+            throw;
+        }
+    }
+
+    public override ResponseHeader Publish(
+        RequestHeader requestHeader,
+        SubscriptionAcknowledgementCollection subscriptionAcknowledgements,
+        out uint subscriptionId,
+        out UInt32Collection availableSequenceNumbers,
+        out bool moreNotifications,
+        out NotificationMessage notificationMessage,
+        out StatusCodeCollection results,
+        out DiagnosticInfoCollection diagnosticInfos)
+    {
+        try
+        {
+            OperationContext context = ValidateRequest(requestHeader, RequestType.CreateSubscription);
+
+            var responseHeader = base.Publish(requestHeader, subscriptionAcknowledgements, out subscriptionId, out availableSequenceNumbers, out moreNotifications, out notificationMessage, out results, out diagnosticInfos);
+
+            int events = 0;
+            int dataChanges = 0;
+            int diagnostics = 0;
+            notificationMessage.NotificationData.ForEach(x =>
+            {
+                if (x.Body is DataChangeNotification changeNotification)
+                {
+                    dataChanges += changeNotification.MonitoredItems.Count;
+                    diagnostics += changeNotification.DiagnosticInfos.Count;
+                }
+                else if (x.Body is EventNotificationList eventNotification)
+                {
+                    events += eventNotification.Events.Count;
+                }
+                else
+                {
+                    Console.WriteLine("Unknown notification type");

```

Comment:
>Console.WriteLine [](http://example.com/codeflow?start=20&length=17)

Why is this a console output and not a log?

Reply:
updated